### PR TITLE
Allow broadcast

### PR DESCRIPTION
**Pull request by shargon**
IBroadcast

### CODE REVIEW

File: src/NeoSharp.Core/DI/Modules/NetworkModule.cs [hunk 1]
```diff
@@ -28,6 +29,8 @@ public void Register(IContainerBuilder containerBuilder)
             containerBuilder.RegisterSingleton<IPeerListener, TcpPeerListener>();
             containerBuilder.RegisterSingleton<ITcpPeerFactory, TcpPeerFactory>();
 
+            containerBuilder.RegisterInstanceCreator<IEnumerable<IBroadcast>>(c => new IBroadcast[] { c.Resolve<IServer>() });
```
Comment (osmirnov) [inline src/NeoSharp.Core/DI/Modules/NetworkModule.cs]: It always contains a single item which is IServer, right?

File: src/NeoSharp.Core/Messaging/IBroadcastMessageHandler[TMessage].cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NeoSharp.Core.ExtensionMethods;
+using NeoSharp.Core.Network;
+
+namespace NeoSharp.Core.Messaging
+{
+    public class IBroadcastMessageHandler<TMessage> : IMessageHandler<TMessage> where TMessage : Message
```
Comment (osmirnov) [inline src/NeoSharp.Core/Messaging/IBroadcastMessageHandler[TMessage].cs]: Drop I preffix: it is class not an interface.

File: src/NeoSharp.Core/DI/Modules/NetworkModule.cs (hunk 1 above)
Comment (osmirnov) [reply src/NeoSharp.Core/DI/Modules/NetworkModule.cs]: My couple thoughts were why to define a collection from a single element and should be broadcast incorporated inside of Server class to propagate messages over connectedPeers collection

File: src/NeoSharp.Core/DI/Modules/NetworkModule.cs [hunk 3]
```diff
@@ -28,9 +29,11 @@ public void Register(IContainerBuilder containerBuilder)
             containerBuilder.RegisterSingleton<IPeerListener, TcpPeerListener>();
             containerBuilder.RegisterSingleton<ITcpPeerFactory, TcpPeerFactory>();
 
+            containerBuilder.RegisterSingleton<IBroadcast, Server>();
```
Comment (osmirnov) [inline src/NeoSharp.Core/DI/Modules/NetworkModule.cs]: Could you please confirm IBroadcast and IServer are pointing to the same instance and do not create Server twice?

Comment (shargon) [reply src/NeoSharp.Core/DI/Modules/NetworkModule.cs]: Confirmed, is the same

image

**Pull request merged**